File: app/src/main/java/com/nhatminh/example/architecture/demoarchitecture/search/view/GithubReposAdapter.java
package com.nhatminh.example.architecture.demoarchitecture.search.view;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.nhatminh.example.architecture.demoarchitecture.R;
import com.nhatminh.example.architecture.demoarchitecture.databinding.RvItemReposBinding;
import com.nhatminh.example.architecture.demoarchitecture.model.GithubRepos;

import java.util.ArrayList;
import java.util.List;

public class GithubReposAdapter extends RecyclerView.Adapter<GithubReposAdapter.ReposViewHolder> {

    List<GithubRepos> reposList;


    public GithubReposAdapter() {
        reposList = new ArrayList<>();
    }

    public GithubReposAdapter(List<GithubRepos> reposList) {
        this.reposList = reposList;
    }

    @NonNull
    @Override
    public ReposViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RvItemReposBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.rv_item_repos, parent, false);
        return new ReposViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ReposViewHolder holder, int position) {
        holder.bind(reposList.get(position));
    }



    @Override
    public int getItemCount() {
        return reposList.size();
    }

    void updateResults(List<GithubRepos> reposList) {
        this.reposList = reposList;
        notifyDataSetChanged();
    }

    static class ReposViewHolder extends RecyclerView.ViewHolder{

        RvItemReposBinding binding;

        public ReposViewHolder(RvItemReposBinding binding) {
            super(binding.getRoot());
            this.binding = binding;

        }

        public void bind(GithubRepos repos){
            binding.setGithubRepos(repos);
            binding.executePendingBindings();
        }
    }
}

File: app/src/main/java/com/nhatminh/example/architecture/demoarchitecture/search/view/GithubReposAdapter.kt
package com.nhatminh.example.architecture.demoarchitecture.search.view

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.nhatminh.example.architecture.demoarchitecture.R
import com.nhatminh.example.architecture.demoarchitecture.databinding.RvItemReposBinding
import com.nhatminh.example.architecture.demoarchitecture.model.GithubRepos

class GithubReposAdapter : RecyclerView.Adapter<GithubReposAdapter.ReposViewHolder>() {

    var reposList : MutableList<GithubRepos> = ArrayList<GithubRepos>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReposViewHolder {
        var rvItemReposBinding :  RvItemReposBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context),
                R.layout.rv_item_repos, parent, false)

        return ReposViewHolder(rvItemReposBinding)
    }

    override fun onBindViewHolder(holder: ReposViewHolder, position: Int) {
        holder.bind(reposList[position])
    }

    override fun getItemCount(): Int = reposList.size

    fun updateResults(reposList : MutableList<GithubRepos>){
        this.reposList = reposList
        notifyDataSetChanged()
    }

    class ReposViewHolder(val binding: RvItemReposBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(repos: GithubRepos){
            binding.githubRepos = repos
            binding.executePendingBindings()
        }
    }
}